pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -238,6 +238,9 @@
             + " affecting the accuracy of the delivery time compared to the scheduled time. Default is 1 second.")
     private long delayedDeliveryTickTimeMillis = 1000;
 
+    @FieldContext(category = CATEGORY_SERVER, doc = "HashedWheelTimer duration tick time for when retrying on delayed delivery")
+    private long delayedDeliveryTickDurationMillis = 1000;
```

Comment:
@baomingyu I don't think we need this config there. For most cases and the current implementation of Pulsar delay message, accuracy in the millisecond level is the expected behavior, If the default value 1000 is not work for you, I think you can decrease it to 5ms or lower. Introduce 2 configurations to control the tick time of the delayed message tracker also will introduce complexity to users to tune the delayed message behavior.

Reply:
Now delayedDeliveryTickTimeMillis is used both delay message and  wheeltimer , so i can not change config only one, 
For decrease delayedDeliveryTickTimeMillis to 5ms or lower, will affect wheeltimer .
Sometimes, delay of several milliseconds is acceptable but not milliseconds ahead，so this configuration ‘delayedDeliveryTickTimeMillis’  can't fully meet the requirements.
 I add this delayedDeliveryTickDurationMillis config to separate the two configurations，after adding delayedDeliveryTickDurationMillis  ,i can config delayedDeliveryTickTimeMillis with 0, and delay message will not be consumer any millisecond level ahead.
Befor changing:
> Befor changing:   
public void initialize(ServiceConfiguration config) {
                this.timer = new HashedWheelTimer(new DefaultThreadFactory("pulsar-delayed-delivery"),
                config.getDelayedDeliveryTickTimeMillis(), TimeUnit.MILLISECONDS);
               this.tickTimeMillis = config.getDelayedDeliveryTickTimeMillis();
    }

> after changing:
 public void initialize(ServiceConfiguration config) {
        this.timer = new HashedWheelTimer(new DefaultThreadFactory("pulsar-delayed-delivery"),
                config.getDelayedDeliveryTickDurationMillis(), TimeUnit.MILLISECONDS);
        this.tickTimeMillis = config.getDelayedDeliveryTickTimeMillis();
    }

Reply:
I know this PR is separating the configuration. I mean the tick time of the HashedWheelTimer also affects accuracy right? If the `delayedDeliveryTickTimeMillis` is 5ms and the `delayedDeliveryTickTimeMillis` is 1000ms, what kind of accuracy guarantee can we get?

Reply:
HashedWheelTimer use delayedDeliveryTickTimeMillis for initial , and InMemoryDelayedDeliveryTracker use delayedDeliveryTickTimeMillis value to check  do or not for delay message in current tick period. when delayedDeliveryTickTimeMillis is 1000ms, delay message will be consumer serval millisecond ahead.
> if (deliveryAt < (now + tickTimeMillis)) {
            // It's already about time to deliver this message. We add the buffer of
            // `tickTimeMillis` because messages can be extracted from the tracker
            // slightly before the expiration time. We don't want the messages to
            // go back into the delay tracker (for a brief amount of time) when we're
            // trying to dispatch to the consumer.
            return false;
        }
And  Introduce 2 configurations to control the tick time of the delayed message and config delayedDeliveryTickTimeMillis with 0 ,will avoid delay message consumed millisecond ahead.

Reply:
@baomingyu Do you have a test for setting delayedDeliveryTickTimeMillis=5 and the delayedDeliveryTickDurationMillis=1000 to check how the delayed message performed?

Reply:
In this moment, the delayedDeliveryTickTimeMillis is used for the HashedWheelTimer, reduce to 5ms will increase the cpu usage. @codelipenghui